shared/components/myPage.js

```diff
@@ -231,6 +230,7 @@ class MyPage extends Component {
           }
           renderRow={(rowData) => <CardLayout dataSource = {rowData}/>}
```

Comment:
I think 225 ~ 234 lines same with 238 ~ 247 lines but one line is different.
'}).cloneWithRows(dataSource)' and '}).cloneWithRows(this.props.createdPosts)'

you can think simple way to reduce line. For example,
let _dataSource = null;
if () _dataSource = dataSource1;
else _dataSource = dataSource2;
return (
  < ... _dataSource ... />
);